specs/protocol/stage-1.md

```diff
@@ -0,0 +1,44 @@
+# Stage 1 Roles and Requirements
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Overview](#overview)
+- [Permissionless Fault Proofs](#permissionless-fault-proofs)
+- [Roles for Stage 1](#roles-for-stage-1)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+This document outlines configuration requirements (including roles and other parameters)
+for an implementation of the OP Stack to satsify the Stage 1 decentralization requirements as defined
+by
+L2 Beat [[1](https://medium.com/l2beat/introducing-stages-a-framework-to-evaluate-rollups-maturity-d290bb22befe), [2](https://medium.com/l2beat/stages-update-security-council-requirements-4c79cea8ef52)].
+
+## Permissionless Fault Proofs
+
+Stage 1 requires a chain to be operating with Permissionless Fault Proofs.
+
+## Roles for Stage 1
+
+Within the context of an OP Stack, the following roles are required for Stage 1:
+
+1. **Upgrade Controller:** ALthough named for its ability to perform an upgarde, more generally this
+   account is in charge of _safety_, meaning it MUST control any action which has an impact
+   on the determination of a valid L2 state, or the custody of bridged assets.
+
+   This includes upgrading L1 contracts, and modifying the implementation of the dispute game, and
+   any other safety-critical function.
+
+2. **Guardian:** This account is in charge of _liveness_, meaning it MUST control any action which
+   may cause a delay in the finalization of L2 states and the resulting settlement on L1.
+
+   This includes but is not limited to pausing code paths related to withdrawals.
+
+There may be additional [roles](./configurability.md#admin-roles) in the system, however they MUST
+not be able to perform any actions which have an impact on either the safety or liveness of the
+system. An example of such a role is the `SystemConfig` owner (AKA Chain Operator), which can
```

Comment:
The system config owner can set the gas fees extremely high, which then affects liveness because no one can afford transactions? Perhaps the right framing is to say these additional roles don't impact the L2 Beat stage 1 requirements, as opposed to safety or liveness

Reply:
You're right. However I've realized that he issue is actually not about liveness, but about allowing withdrawals. I will update this text to clarify that.


In order for withdrawals to continue, we need:
1. force inclusion (deposit tx) to work. This can be blocked by either:
  - resource config params being outside of a reasonable range, which is why all [these checks exist](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/src/L1/SystemConfig.sol#L446-L461).
  - gas limit params being outside of a reasonable range (prevented by [these checks](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/src/L1/SystemConfig.sol#L409-L410)).
2. permissionless output proving
3. withdrawals being unpaused

Since 1 is enforced in the contract, and 2 and 3 are guardian actions that's all good. 

As you note, there is also the [`setGasConfig`](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/src/L1/SystemConfig.sol#L363-L398) logic which the chain operator has access to, however these values do not impact on the deposit path, only sequenced tx, so they cannot censor withdrawals.

Reply:
This document is now written with more precise language, so that instead of 'safety/liveness', it focuses on the state transition validity vs. withdrawals.

Accordingly, I think we can resolve this issue.